src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -190,14 +190,26 @@ export function useDiscoverState({
          *  That's because appState is updated before savedSearchData$
          *  The following line of code catches this, but should be improved
          */
-        const nextDataView = await loadDataView(
+        const nextDataViewData = await loadDataView(
           services.dataViews,
           services.uiSettings,
           nextState.index
         );
-        savedSearch.searchSource.setField('index', nextDataView.loaded);
+        const nextDataView = resolveDataView(
+          nextDataViewData,
+          savedSearch.searchSource,
+          services.toastNotifications
+        );
 
-        reset();
+        // If the requested data view is not found, don't try to load it,
+        // and instead reset the app state to the fallback data view
+        if (nextDataViewData.stateValFound) {
+          savedSearch.searchSource.setField('index', nextDataView);
+          reset();
+        } else {
+          setAppState({ index: nextDataView.id });
+          return;
+        }
```

Comment:
Since this part will soon be refactored, we should think of simplify the logic a bit, let's focus on the fact that, there's no user feedback when an invalid data view id is entered, so maybe just like this

```suggestion
        // If the requested data view is not found, don't try to load it,
        // instead provide feedback to the user
        if (!nextDataViewData.stateValFound) {
          return;
        }
        
        savedSearch.searchSource.setField('index', nextDataView);
        reset();
```
on the other hand: what would be the advantage of setting the state to the default index, given that the manipulation of the URL that leads to this error is usually done by one of us developers?

Reply:
It shouldn't get set back to the default data view, it should stay on the one the user was viewing before changing the URL, since that's the data view that `resolveDataView` will return as the fallback.

The issue with not calling `setAppState({ index: nextDataView.id });` is that the app state has already been updated to include the invalid data view ID, so if we don't change it back, the app state and URL will be out of sync with the data view we show to the user. In my testing it caused weird things like broken share URLs, etc. since they would be using the invalid data view ID from app state.

But you're right all of this only happens when someone manually changes the URL from what I can tell, so probably not very common.

Reply:
Thx, LGTM 👍